packages/test-utils/src/reporter.ts

```diff
@@ -37,31 +41,39 @@ interface TestRunTestInputModel {
   recordingIds: string[];
 }
 
-export interface TestIdContext {
-  title: string;
-  scope: string[];
-  attempt: number;
-}
-
 export type UploadStatusThreshold = "all" | "failed-and-flaky" | "failed";
 
 type UploadStatusThresholdInternal = UploadStatusThreshold | "none";
 
 export type UploadOption =
   | boolean
   | {
+      /**
+       * Minimize the number of recordings uploaded for a test attempt (within a shard).
+       * e.g. Only one recording would be uploaded for a failing test attempt, regardless of retries.
+       * e.g. Two recordings would be uploaded for a flaky test attempt (the passing test and one of the failures).
+       */
       minimizeUploads?: boolean;
       statusThreshold?: UploadStatusThreshold;
     };
 
-interface UploadableTestRunResult<TRecordingMetadata extends UnstructuredMetadata> {
+interface UploadableTestExecutionResult<TRecordingMetadata extends UnstructuredMetadata> {
+  executionGroupId: string;
   attempt: number;
   maxAttempts: number;
   recordings: RecordingEntry<TRecordingMetadata>[];
   result: TestRun["result"];
   testId: string;
 }
 
+interface UploadableTestResult<TRecordingMetadata extends UnstructuredMetadata> {
+  executions: Record<string, UploadableTestExecutionResult<TRecordingMetadata>[]>;
+  uploadedStatuses: {
+    passed: boolean;
+    failed: boolean;
```

Comment:
I don't think I understand what this is supposed to mean.

Reply:
this is meant to track what has been uploaded for flaky tests since those flakes can be in different execution groups

Reply:
I see. I think the naming is kind of confusing to me. It seemed _prescriptive_ (aka "you should upload recordings with this status") rather than _descriptive_ (aka "I already uploaded recordings with this status"), maybe because I read "upload statuses" instead of "upload**ed** statuses" but I think that might be an easy mistake to make. Maybe "did upload statuses" would be a tiny bit clearer?